test/serializer_test.exs

```diff
@@ -25,6 +25,39 @@ defmodule JSONAPISerializerTest do
     end
   end
 
+  # The `metadata` field may contain arbitrary key-value data, such as user-supplied JSON.
+  defmodule PostWithMetadataView do
+    use JSONAPI.View
+
+    def fields,
+      do: [
+        :text,
+        :body,
+        :full_description,
+        :inserted_at,
+        :metadata,
+        :_private_attribute,
```

Comment:
The reason I created a new view was so that the "basic" tests could keep using the simpler view, and I wanted to make as small of an impact on existing tests as I could. But I agree, this view duplicates a lot. I'd be happy to merge them if that's preferred